test/unit/cmap/auth/mongodb_oidc/token_machine_workflow.test.ts

```diff
@@ -0,0 +1,34 @@
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+
+import { Connection, MongoCredentials, TokenMachineWorkflow } from '../../../../mongodb';
+
+describe('TokenMachineFlow', function () {
+  describe('#execute', function () {
+    const workflow = new TokenMachineWorkflow();
+
+    context('when OIDC_TOKEN_FILE is not in the env', function () {
+      let file;
+      const connection = sinon.createStubInstance(Connection);
+      const credentials = sinon.createStubInstance(MongoCredentials);
+
+      before(function () {
+        file = process.env.OIDC_TOKEN_FILE;
+        delete process.env.OIDC_TOKEN_FILE;
+      });
+
+      after(function () {
+        process.env.OIDC_TOKEN_FILE = file;
```

Comment:
You cannot unconditionally assign back to process.env without potentially setting this to a non-empty string which can have side-effects on other tests. if OIDC_TOKEN_FILE is `undefined` then `process.env.OIDC_TOKEN_FILE` will be set to `"undefined"` the string.

Reply:
Nice catch, I've added a check if `file` exists before setting it back.